### PR TITLE
Bugfix FXIOS-4251 [v117] Sponsored tiles don't show based on defaultSearchEngine

### PR DESCRIPTION
**Pull request by nishant2718**
## :scroll: Tickets
[FXIOS-4251](https://mozilla-hub.atlassian.net/browse/FXIOS-4251)
https://github.com/mozilla-mobile/firefox-ios/issues/10773

## :bulb: Description

Sponsored tiles don't show in TopSites section if it matches the default search engine. 
## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and ensured the tests suite is passing
- [ ] Implemented accessibility and tested on UI related work (minimum Dynamic Text and VoiceOver)
- [ ] Updated documentation / comments for complex code and public methods if needed

### CODE REVIEW

File: Client/Frontend/Home/TopSites/DataManagement/SponsoredTileDataUtility.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ struct SponsoredTileDataUtility: SponsoredTileDataUtilityInterface {
         guard let defaultSearchEngine = searchEngine,
               let secondLevelDomain = site.secondLevelDomain else { return true }
 
-        let hasMatchedEngine = defaultSearchEngine.shortName.caseInsensitiveCompare(secondLevelDomain) == .orderedSame
+        let hasMatchedEngine = defaultSearchEngine.shortName.localizedCaseInsensitiveContains(secondLevelDomain)
```
Comment (lmarceau) [inline Client/Frontend/Home/TopSites/DataManagement/SponsoredTileDataUtility.swift:17]: Can we add a test for this, so we cover that we do the right thing when we need that `contains`?

File: Tests/ClientTests/Frontend/Home/TopSites/TopSitesDataAdaptorTests.swift [hunk 2]
```diff
@@ -598,11 +596,7 @@ extension TopSitesDataAdaptorTests {
     }
 
     func add(searchEngine: OpenSearchEngine) {
-        mockSearchEngineProvider = MockSearchEngineProvider()
```
Comment (lmarceau) [inline Tests/ClientTests/Frontend/Home/TopSites/TopSitesDataAdaptorTests.swift:601]: I think we can remove the test variables `mockSearchEngineProvider` and `searchEngines` then?

File: Client/Frontend/Home/TopSites/DataManagement/SponsoredTileDataUtility.swift (hunk 1 above)
Comment (nishant2718) [reply Client/Frontend/Home/TopSites/DataManagement/SponsoredTileDataUtility.swift:17]: It looks like `testShouldAdd_filterSite_whenCaseInsentiveIsUsed` is already covering the case when the default search engine's short name matches the SLD of a site. 

Is there something additional you're looking for?

Comment (lmarceau) [reply Client/Frontend/Home/TopSites/DataManagement/SponsoredTileDataUtility.swift:17]: Oh that's for the case insensitive (so `TeSt` website will be filtered even when the search engine is `test`). The extra test case I am mentioning is to have a `test.com` website filtered when the search engine is `test`. So we make sure we don't check for equality, but with contains.

**Review by lmarceau — APPROVED** [review]

**Comment by nishant2718** [comment]
BR is ✅ 

image

**Pull request merged**